frontend/src/navigations/StackNavigation.js

```diff
@@ -1,11 +1,32 @@
-import React from "react";
-import { createStackNavigator } from "@react-navigation/stack";
-import TabNavigation from "./TabNavigation";
+import React from 'react';
+import { createStackNavigator } from '@react-navigation/stack';
+import { BackHandler, Alert } from 'react-native';
+import { useFocusEffect } from '@react-navigation/native';
+import TabNavigation from './TabNavigation';
 import SplashScreen from '../screens/SplashScreen';
 
 const Stack = createStackNavigator();
 
-const StackNavigation = () => {
+const StackNavigation = ({ navigation, route }) => {
```

Comment:
@sidharth2829  your problem lies in this line, see here you passing  ({navigation, route}) at the function definition, but you haven't changed your App. tsx file, where you call this function There it looks like.

![image](https://github.com/SB2318/UltimateHealth/assets/87614560/0b80581d-913e-4dfa-a420-2700a3ab4b86)

You are not sending the props there.

In that case, My suggestion is to call the **StackNavigation** component properly,

or you can use  **useRoute** hook from 'react-navigation/native'

code example:

import { useRoute } from '@react-navigation/native';

```
const StackNavigation =  ()=>{
  const route = useRoute();

  // Access route properties
  console.log(route.params);

  return (
    // rest of your code
  );
}

```

Reply:
> @sidharth2829 your problem lies in this line, see here you passing ({navigation, route}) at the function definition, but you haven't changed your App. tsx file, where you call this function There it looks like.
> 
> ![image](https://private-user-images.githubusercontent.com/87614560/335249477-0b80581d-913e-4dfa-a420-2700a3ab4b86.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcwODIwOTAsIm5iZiI6MTcxNzA4MTc5MCwicGF0aCI6Ii84NzYxNDU2MC8zMzUyNDk0NzctMGI4MDU4MWQtOTEzZS00ZGZhLWE0MjAtMjcwMGEzYWI0Yjg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMwVDE1MDk1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4NzFiZjQ4ZDhhYTJiZGQxMTEwMTEzNThmNjdhMzg4OGUxYzFmMjk5ZDAwZmRhYjc2NjRmMjllNjNjZmIxMjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.V2FgWxms4x9hWm-Iw0CI1dDYfoVJ6ShzCMW2Ke6VfLA)
> 
> You are not sending the props there.
> 
> In that case, My suggestion is to call the **StackNavigation** component properly,
> 
> or you can use **useRoute** hook from 'react-navigation/native'
> 
> code example:
> 
> import { useRoute } from '@react-navigation/native';
> 
> ```
> const StackNavigation =  ()=>{
>   const route = useRoute();
> 
>   // Access route properties
>   console.log(route.params);
> 
>   return (
>     // rest of your code
>   );
> }
> ```

Ok, i will make appropriate changes and i have messaged you on discord. Can you help to run it on android as well?

Reply:
I haven't receive, but it will resolve your issue for android also.